src/safeposix/syscalls/fs_calls.rs

```diff
@@ -26,33 +51,48 @@ impl Cage {
     }
 
     pub fn open_syscall(&self, path: &str, flags: i32, mode: u32) -> i32 {
-        //Check that path is not empty
+        // Check that the given input path is not empty
         if path.len() == 0 {
             return syscall_error(Errno::ENOENT, "open", "given path was null");
         }
+        
+        // Retrieve the absolute path from the root directory. The absolute path is then used to validate directory paths
+        // while navigating through subdirectories and creating a new file or open existing file at the given location.
         let truepath = normpath(convpath(path), self);
 
+        // Fetch the next file descriptor and its lock write guard to ensure the file can be associated with the file descriptor 
         let (fd, guardopt) = self.get_next_fd(None);
+        // Return an error when no file descriptor is available
         if fd < 0 {
```

Comment:
Will the same error value be returned for all error types?  should you check for a certain error return value and panic if you get something else?

Reply:
I checked the implementation for this function call (`get_next_fd`) and whenever there is success, it returns a value > 0. But in the remaining cases, including the error, we always get the same negative value returned from this function. So, that's why I think this implementation works fine.
I'll update the comment in the code for better clarity.

Reply:
In general, it's a good idea to assert or similar in a situation like this because if they change that function later, your code would behave in unexpected ways.

Reply:
Understood, have added matching checks for it. Can you please verify if this is the expected way? Or any other thoughts on the approach would be great.